ExcelWakaTimeAddin/WakaTimeAddIn.cs

```diff
@@ -125,6 +125,68 @@ private static void PromptApiKey()
             form.ShowDialog();
         }
 
+        private static string GetFileLocalPath(string rawPath)
+        {
+            try
+            {
+                var workbookUri = new Uri(rawPath);
+                // If local file, return it as-is
+                if (workbookUri.IsFile)
+                    return rawPath;
+
+                string localPath = string.Empty;
+                // Registry key names to loop                
+                System.Collections.Generic.List<string> keyNames = new System.Collections.Generic.List<string>() { "OneDrive", "OneDriveCommercial", "OneDriveConsumer", "onedrive" };
+                foreach (var keyName in keyNames)
+                {
+                    using (var key = Microsoft.Win32.Registry.CurrentUser.OpenSubKey("Environment", false))
+                    {
+                        var rootDirectory = key.GetValue(keyName).ToString();
+                        if (string.IsNullOrEmpty(rootDirectory) == false && System.IO.Directory.Exists(rootDirectory))
+                        {
+                            if (keyName == "OneDrive")
```

Comment:
This is duplicated, could be moved into a function and called in the two places instead of repeating the code.